src/test/java/org/mifos/integrationtest/cucumber/stepdef/ReportManagementSteps.java

```diff
@@ -0,0 +1,312 @@
+        reportDTO.setReportType("Financial");
+        reportDTO.setReportSubType("Monthly");
+        reportDTO.setReportCategory("Operational");
+        reportDTO.setDescription("Financial summary for the month of June");
+        reportDTO.setReportSql("SELECT * FROM transactions WHERE date >= '2024-06-01' AND date <= '2024-06-30'");
+        List<ReportParameter> reportParameters = new ArrayList<>();
+
+        ReportParameter parameter1 = new ReportParameter();
+        parameter1.setParameterKey("Param1");
+        parameter1.setParameterValue("Value1");
+
+        ReportParameter parameter2 = new ReportParameter();
+        parameter2.setParameterKey("Param2");
+        parameter2.setParameterValue("Value2");
+
+        reportParameters.add(parameter1);
+        reportParameters.add(parameter2);
+
+        reportDTO.setReportParameters(reportParameters);
+
+        try {
+            scenarioScopeState.createReportBody = objectMapper.writeValueAsString(reportDTO);
+        } catch (JsonProcessingException e) {
+            logger.error("Unable to convert the DTO : {}", e);
+        }
+
+        Awaitility.await().atMost(awaitMost, SECONDS).pollDelay(pollDelay, SECONDS).pollInterval(pollInterval, SECONDS).untilAsserted(() -> {
+            RequestSpecification requestSpec = Utils.getDefaultSpec();
+            scenarioScopeState.response = RestAssured.given(requestSpec).header("Content-Type", "application/json")
+                    .header("Platform-TenantId", scenarioScopeState.tenant).baseUri(operationsAppConfig.operationAppContactPoint)
+                    .body(scenarioScopeState.createReportBody).expect().statusCode(expectedStatus).when()
+                    .post(operationsAppConfig.reportCreate).andReturn().asString();
+
+            logger.info("Create Report Response: {}", scenarioScopeState.response);
+
+            ObjectMapper responseMapper = new ObjectMapper();
+            try {
+                JsonNode jsonResponse = responseMapper.readTree(scenarioScopeState.response);
+                scenarioScopeState.reportId = jsonResponse.get("id").asText();
+            } catch (Exception e) {
+                logger.error("Error parsing JSON response to get report ID: {}", e.getMessage());
+                fail("Error parsing JSON response to get report ID");
+            }
+        });
+    }
+
+    @Then("the response should contain the created report details")
+    public void theResponseShouldContainCreatedReportDetails() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            JsonNode jsonResponse = objectMapper.readTree(scenarioScopeState.response);
+
+            MatcherAssert.assertThat(jsonResponse.get("reportName").asText(), Matchers.equalTo("Sample Report"));
+            MatcherAssert.assertThat(jsonResponse.get("reportType").asText(), Matchers.equalTo("Financial"));
+            MatcherAssert.assertThat(jsonResponse.get("reportSubType").asText(), Matchers.equalTo("Monthly"));
```

Comment:
Instead of using the MatcherAssert class repeatedly, use its functions directly.